### PR TITLE
Feature: allow to disable stock management

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
Closes https://github.com/mirumee/saleor/issues/1979.

This adds a new field `track_inventory` on the variant model, which toggles to inventory management (allocations and checks).

This also adds a new settings (on the SiteSettings model), that allows to set the default inventory tracking for the variant creation form.

It's only there to allow users to save time and prevent them from accidentally setting an unwanted inventory management.

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: saleor/product/models.py [hunk 1]
```diff
@@ -185,6 +185,11 @@ class ProductVariant(models.Model):
         Product, related_name='variants', on_delete=models.CASCADE)
     attributes = HStoreField(default={}, blank=True)
     images = models.ManyToManyField('ProductImage', through='VariantImage')
+    handle_stock = models.BooleanField(
```
Comment (elwoodxblues) [inline saleor/product/models.py]: Although I understood the purpose of this field from reading the code, this name sounds a bit too general to me. What about `track_inventory`? I think it would be more understandable.

File: saleor/product/models.py [hunk 2]
```diff
@@ -185,6 +185,11 @@ class ProductVariant(models.Model):
...
+        default=True,
+        help_text=pgettext_lazy(
+            'product variant handle stock field help text',
+            'Allow or not to automatically handle stock units from orders'))
```
Comment (elwoodxblues) [inline saleor/product/models.py]: We tend to keep help texts in forms rather than in models. Also, if we changed `handle_stock` to `track_inventory` we would have to rephrase it to something like: "Automatically track this product's inventory".

File: saleor/site/models.py [hunk 3]
```diff
@@ -22,6 +22,12 @@ class SiteSettings(models.Model):
     include_taxes_in_prices = models.BooleanField(default=True)
     display_gross_prices = models.BooleanField(default=True)
     charge_taxes_on_shipping = models.BooleanField(default=True)
+    handle_stock_by_default = models.BooleanField(
+        default=True,
+        help_text=pgettext_lazy(
```
Comment (elwoodxblues) [inline saleor/site/models.py]: Again, this help_text should be moved to the form. I think we need to explain precisely what this feature does, in a user-friendly way. I'm thinking about a checkbox with descriptive label and help_text:
- label: "Enable inventory tracking for newly created products."
- help text: "Inventory tracking may be modified at the product-level. Changing this setting won't affect existing products."

What do you think?

**Comment by NyanKiyoshi** [comment]
@elwoodxblues all good for me, I will do the changes. Thanks!
